themes/default/content/docs/intro/concepts/stack.md

```diff
@@ -700,6 +700,275 @@ as your current stack (i.e. when deploying the "staging" stack of the above prog
+
+{{% choosable language javascript %}}
+
+```javascript
+const infra = new pulumi.StackReference(...);
+const ip = infra.getOutput("privateIp");
+const logKey = ip.apply(ip => `logs/${ip}.log`);
+const logFile = new aws.s3.BucketObject("log", {
+    // ...
+    key: logKey
+});
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const infra: StackReference = new pulumi.StackReference(...);
+const ip: Output<string> = infra.getOutput("privateIp");
+const logKey: Output<string> = ip.apply(ip => `logs/${ip}.log`);
+const logFile: aws.s3.BucketObject = new aws.s3.BucketObject("log", {
+    // ...
+    key: logKey
+});
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+infra = StackReference(...)
+ip = infra.get_output("privateIp")
+log_key = ip.apply(lambda ip: f"logs/{ip}.log")
+log_file = aws.s3.BucketObject("log", {
+    # ...
+    key: log_key
+})
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+infra, err := pulumi.NewStackReference(ctx, ...)
+if err != nil {
+    return err
+}
+ip := infra.GetOutput("privateIp")
+logKey := ip.ApplyT(func(ip string) string {
+    return fmt.Sprintf("logs/%s.log", ip)
+}).(StringOutput)
+logFile := s3.NewBucketObject(ctx, "log", &s3.BucketObjectArgs{
+    // ...
+    Key: logKey,
+})
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var infra = new StackReference(...);
+var ip = infra.GetOutput("privateIp");
+var logKey = ip.Apply(ip => $"logs/{ip}.log");
+var logFile = new Aws.S3.BucketObject("log", new Aws.S3.BucketObjectArgs
+{
+    // ...
+    Key = logKey,
+});
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+StackReference infra = new StackReference(...);
+Output<String> ip = infra.getOutput("privateIp");
+Output<String> logKey = ip.apply(ip -> String.format("logs/%s.log", ip));
+BucketObject logFile = new BucketObject("log", new BucketObjectArgs.Builder()
+    .key(logKey)
+    .build());
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+On the other hand, as an example of using **`getOutputDetails`**,
+suppose that your referenced stack creates a VPC
+and exports a list of public subnets as a JSON-serialized string,
+and you want to add a bastion host to each subnet.
+With `getOutputDetails`, this would look something like this:
+
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+const infra = new pulumi.StackReference(...);
+const subnetsJSON = await infra.getOutputDetails("subnets");
+const subnets = JSON.parse(subnetsJSON.value);
+for (let i = 0; i < subnets.length; i++) {
+    const subnet = subnets[i];
+    const host = new aws.ec2.Instance(`bastion-${i}`, {
+        // ...
+        subnetId: subnet.id,
+    });
+    // ...
+}
+```
+
+Note that your Pulumi program must export a top-level `async` function
+to be able to use the `await` operator.
```

Comment:
how important is this note? should it be using the shortcode?

Reply:
Now that I think about it, it's not _that_ important.
Yeah, that can drop the short code.

Reply:
🤦  Derp, sorry, you were suggesting that this could use the shortcode.
I thought I'd used it and you were advising against it.

So, to expand on why it's not super important:
code will fail to compile loudly if they don't do this. It's mentioned in the Pulumi Getting Started docs for Typescript.
I replicated it here mostly because this API is one of the places where they'll actually need to use an `await` in the Pulumi program, so it's good to mention "FYI, this is what you'll need if it doesn't work."